### PR TITLE
doc: add external ref

### LINKED ISSUE #2325: status.externalRef vs spec.resourceID
Context: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2312/files/eca4722eac14047ed5e0879cdd89f313bdbc9d44#r1685166296

We should decide if we want to stop populating spec.resourceID and instead populate status.externalRef for new resources and alpha resources.

I _think_ we have agreed this, we just need to write it down

/assign @yuwenma 
cc @acpana

### PR DESCRIPTION
**Pull request by yuwenma**
### Change description

Fixes #2325 

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: docs/develop-resources/api-conventions/external-reference.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# TL;DR
+
+`status.externalRef` should be used as the GCP identifier of the Config Connector resource.
+
+
+# Basic Rules
+
+Config Connector should assign the `status.externalRef` value when it firstly acquires or creates the GCP resource.
```
Comment (justinsb) [inline docs/develop-resources/api-conventions/external-reference.md:11]: Nit: s/firstly/initially

File: docs/develop-resources/api-conventions/external-reference.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+![reconciliation](./image/externalref-reconcile.png)
+
+
+# API definition
+
+```
+type <Kind>Status struct {
+   /* A unique specifier for the <Kind> workerpool resource in GCP.*/
+   ExternLRef *string json:"externalRef,omitempty"
```
Comment (justinsb) [inline docs/develop-resources/api-conventions/external-reference.md:29]: Nit: typo in field name

File: docs/develop-resources/api-conventions/external-reference.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+   ...
+}
+```
+
+## `status.externalRef `format
```
Comment (justinsb) [inline docs/develop-resources/api-conventions/external-reference.md:34]: Nit: misplaced space

File: docs/develop-resources/api-conventions/external-reference.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+## Enhancement to the service-generated ID resource
+
+* If a resource does not let user to specify the name or identifier, but its GCP service assigns an ID in the HTTP response, the Direct reconciler will use the service generated ID to build the `status.externalRef`, but **not write back to the `spec.resourceID`**. 
+
+* If user specifies the service-generated ID in the `spec.resourceID` field, the Direct reconciler will acquire the resource if the `status.externalRef` is not assigned.
+
+### Backward compatibiltiy for TF/DCL based reconcilers
+
+The existing TF-based and DCL-based reconciliation treats the `spec.resourceID` as the single Source of True to identify a GCP resource. It writes back the `spec.resourceID` after reconciliation. Here are [the known problems](https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/2765). Config Connector will keep the bahvior of these resources, until they are migrated to the Direct reconciler.
```
Comment (justinsb) [inline docs/develop-resources/api-conventions/external-reference.md:49]: Nit: "Source of Truth"

Comment (justinsb) [reply docs/develop-resources/api-conventions/external-reference.md:49]: Also typo in behavior

**Comment by justinsb** [comment]
LGTM, a few spelling nits!  Do you want to address in this PR or in follow on?

File: docs/develop-resources/api-conventions/external-reference.md (hunk 1 above)
Comment (yuwenma) [reply docs/develop-resources/api-conventions/external-reference.md]: Done!

File: docs/develop-resources/api-conventions/external-reference.md (hunk 2 above)
Comment (yuwenma) [reply docs/develop-resources/api-conventions/external-reference.md]: ahhh.

File: docs/develop-resources/api-conventions/external-reference.md (hunk 3 above)
Comment (yuwenma) [reply docs/develop-resources/api-conventions/external-reference.md]: Yeah, this is from "Docs™ to Markdown" extension, somehow it always misplaces the "`"

File: docs/develop-resources/api-conventions/external-reference.md (hunk 4 above)
Comment (yuwenma) [reply docs/develop-resources/api-conventions/external-reference.md]: Thanks!

**Review by justinsb — APPROVED** [review]

**Pull request merged**